plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/WebFragmentWarPublishTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.integration.appengine;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import com.google.cloud.tools.eclipse.appengine.deploy.WarPublisher;
+import com.google.cloud.tools.eclipse.test.util.ZipUtil;
+import com.google.cloud.tools.eclipse.test.util.project.ProjectUtils;
+import java.io.IOException;
+import java.util.List;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IFolder;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.junit.After;
+import org.junit.Test;
+
+public class WebFragmentWarPublishTest {
+
+  private final IProgressMonitor monitor = new NullProgressMonitor();
+  private IProject project;
+
+  @After
+  public void tearDown() throws CoreException {
+    if (project != null) {
+      ProjectUtils.waitForProjects(project);
+      project.delete(true, null);
+    }
+  }
+
+  @Test
+  public void testPublishExploded_webFragmentJarPublished() throws IOException, CoreException {
+    List<IProject> projects = ProjectUtils.importProjects(getClass(),
+        "test-projects/web-fragment-example.zip", false /* checkBuildErrors */, monitor);
```

Comment:
Could you change the `spring-web` version to _4.3.8.RELEASE_ to match the `spring-boot-appengine` test project?  Saves downloading two versions of the same artifact.